src/portal-impl.c

```diff
@@ -103,13 +103,51 @@ register_portal (const char *path, gboolean opt_verbose, GError **error)
   return TRUE;
 }
 
+static gboolean
+g_strv_case_contains (const gchar * const *strv,
+                      const gchar         *str)
+{
+  for (; *strv != NULL; strv++)
+    {
+      if (g_ascii_strcasecmp (str, *strv) == 0)
+        return TRUE;
+    }
+
+  return FALSE;
+}
+
 static gint
-sort_impl_by_name (gconstpointer a,
-                   gconstpointer b)
+sort_impl_by_use_in (gconstpointer a,
```

Comment:
Since you're still comparing by name in the fallback case, `sort_impl_by_use_in_and_name` seems more appropriate